### PR TITLE
Print more context when FileWatching test fails; also make small tweaks to the `ContextTestSet` (`@testset let ...`) functionality in the Test stdlib

### PR DESCRIPTION
**Pull request by Keno**
Follow-up to https://github.com/JuliaLang/julia/pull/46138

### CODE REVIEW

**Comment by DilumAluthge** [comment]
In the case where the left-hand side (`name`) is an expression (instead of a symbol), should we throw an error if `name.head != :tuple`? Since presumably we don't intend to (currently) support any other usage?

**Comment by DilumAluthge** [comment]
On one hand, I guess it doesn't really hurt to let users stick whatever they want on the left-hand side.

But on the other hand, we might (in the future) want to have different behavior based on what the LHS expression is, and in that case, we should make it error now, so that it's easier for us to add those features later.

**Comment by Keno** [comment]
I'd be fine with making it an error, but I also think we can just change the behavior in the future, since it should only affect printing.

File: stdlib/Test/src/Test.jl [hunk 1]
```diff
@@ -971,16 +971,16 @@ about a context object that is being tested.
...
     parent_ts::AbstractTestSet
-    context_sym::Symbol
+    context_name::Union{Symbol, Expr}
     context::Any
 end
 
-function ContextTestSet(sym::Symbol, @nospecialize(context))
-    ContextTestSet(get_testset(), sym, context)
+function ContextTestSet(name::Union{Symbol, Expr}, @nospecialize(context))
+    ContextTestSet(get_testset(), name, context)
```
Comment (DilumAluthge) [inline stdlib/Test/src/Test.jl:979]:
```suggestion
    if (name isa Expr) && (name.head != :tuple)
        error("Invalid syntax: $(name)")
    end
    return ContextTestSet(get_testset(), name, context)
```

**Review by DilumAluthge — APPROVED** [review]

**Pull request merged**